### PR TITLE
Feature: Introduce Github Actions for sample program

### PR DESCRIPTION
**Pull request by jjmilburn**
## Contribution Guideline Acceptance

- [X ] I have read and understand the [Individual Contributors License Agreement](CONTRIBUTING.md).
- [X ] I accept the [Individual Contributors License Agreement](CONTRIBUTING.md) on the entire contents of this pull request.

## Proposed Changes

Introduce `expect` scripts which automatically build and run the desktop sample program. The tests introduced here are meant to ensure that the behavior explained in the `README` for the sample program always works as described.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [X ] I have added unit tests with >= 90% coverage for all files modified or added by these changes
- [X ] I have designed my tests to prove that my fix is effective or that my feature works as described
- [ ] I have added docstrings to all new functions describing their purpose, parameters, and return value
- [ ] I have updated or added supplemental documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further Comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

Link to any supplemental information here.

### Reviewers

This will be filled by the people who review this particular Pull Request.

### Attribution

This PR template derived from:
https://github.com/skcript/PR-Template

Original template is:

Copyright (c) 2017 Skcript

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

### CODE REVIEW

File: .github/workflows/build-and-run-example-desktop-program.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      uses: actions/checkout@v2
+    - name: Install dependencies and ${{ matrix.gcc-version }}
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y ${{ matrix.gcc-version }}
+        sudo apt-get install -y expect  # for testing executable
+        make install-dependencies
+    - name: Build Executable
+      run: |
+        # Run twice to ensure that the default make action is `clean all`
```
Comment (ricehornet) [inline .github/workflows/build-and-run-example-desktop-program.yml:26]: Don't understand why we need to run twice -- doesn't our default make command run `clean all` (as per the Makefile)?

**Review by ricehornet — APPROVED** [review]
Looks good to me, once you're satisfied with artifacts.

File: .github/workflows/build-and-run-example-desktop-program.yml (hunk 1 above)
Comment (jjmilburn) [reply .github/workflows/build-and-run-example-desktop-program.yml:26]: it does - but the first time, there is nothing to clean. This is just a sanity check that running `make clean all` more than once doesn't blow up for any reason.

**Comment by jjmilburn** [comment]
Will wait to merge as we want the other (unrelated) changes in this PR to be mirrored over first.

**Pull request merged**